packages/flutter_tools/lib/src/android/migrations/gradle_java_version_conflict_migration.dart

```diff
@@ -0,0 +1,72 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../../base/file_system.dart';
+import '../../base/project_migrator.dart';
+import '../../base/version.dart';
+import '../../project.dart';
+import '../android_studio.dart';
+
+final Version _androidStudioFlamingo = Version(2022, 2, 0);
```

Comment:
```suggestion
// Android Studio 2022.2 "Flamingo" is the first to bundle a Java 17 JDK.
// Previous versions bundled a Java 11 JDK.
final Version _androidStudioFlamingo = Version(2022, 2, 0);
```

(Or that comment could go somewhere else, like next to the link to the Gradle-Java compatibility matrix. But I think it'd be good to get that piece of information down in a comment here, because that's the the key other piece of data needed for understanding why Flamingo and how that connects to the other numbers in this code.)